std/stdio.d

```diff
@@ -375,10 +375,24 @@ struct File
 
     private struct Impl
     {
+        import std.typecons : BitFlags;
+
         FILE * handle = null; // Is null iff this Impl is closed by another File
         shared uint refs = uint.max / 2;
         bool isPopened; // true iff the stream has been created by popen()
-        Orientation orientation;
+        Orientation orientation; 
+
+        enum OpenMode
+        {
+            r = 1 << 0,
+            w = 1 << 1,
+            a = 1 << 2,
+            b = 1 << 3,
+            // +
+            p = 1 << 4
+        }
+
+        BitFlags!OpenMode openModeFlags;
```

Comment:
- Make it `version(Windows)` (it's only used on windows after all)
- `BitFlags` is redundant given your enum definition
- `openModeFlags` should be placed after `isPopened` for better packing.

Reply:
Not that it matters anymore, but isn't the whole point of `BitFlags` to provide a safer interface than the raw enum value? If that's the case, I don't think it's superfluous.